### PR TITLE
feature(lint) @roots/eslint-config

### PR DESCRIPTION
**Pull request by kellymears**
## Overview

New package: `@roots/eslint-config`. 

eslint docs on what we're doing: https://eslint.org/docs/developer-guide/shareable-configs

By following the eslint standard users don't have to resolve anything in .eslintrc. 

Example:

```
module.exports = {
  root: true,
  extends: ['@roots/eslint-config/sage'],
};
```

All the configs are compositional, so `sage` is at the highest level and is the equivalent of:

```
module.exports = {
  root: true,
  extends: ['@roots/eslint-config/wordpress', '@roots/eslint-config/recommended'],
};
```

If `@roots/eslint-config` is used it falls back on a set of common defaults:

```
module.exports = {
  extends: ['eslint:recommended'],
  plugins: ['import'],
  env: {
    browser: true,
    es6: true,
    node: true,
  },
  parserOptions: {
    sourceType: 'module',
    ecmaFeatures: {
      globalReturn: false,
      jsx: true,
    },
    ecmaVersion: 2020,
  },
}
```

All of this is just js in the root of the package otherwise eslint has a hard time finding it and/or users have to specify the full path relative to the package root 👎🏼

refers: none
closes: none

### CODE REVIEW

File: sources/@roots/eslint-config-bud/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  <img src="https://cdn.roots.io/app/uploads/logo-bud.svg" height="100" alt="Bud" />
+</p>
+
+<p align="center">
+  <img alt="MIT License" src="https://img.shields.io/github/license/roots/bud?color=%23525ddc&style=flat-square" /> <img alt="npm" src="https://img.shields.io/npm/v/@roots/bud.svg?color=%23525ddc&style=flat-square" /> <img alt="Follow Roots" src="https://img.shields.io/twitter/follow/rootswp.svg?color=%23525ddc&style=flat-square" />
+</p>
+
+<h1 align="center">
+  <strong>@roots/bud-typescript</strong>
+</h1>
```
Comment (kellymears) [inline sources/@roots/eslint-config-bud/README.md]: this readme will be overwritten on next docs gen

File: sources/@roots/eslint-config-bud/package.json [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  },
+  "types": "./types/index.d.ts",
+  "devDependencies": {
+    "@roots/bud-babel": "workspace:sources/@roots/bud-babel",
+    "@roots/bud-framework": "workspace:sources/@roots/bud-framework",
+    "@skypack/package-check": "0.2.2",
+    "@types/eslint": "8.2.2",
+    "@types/lodash": "4.14.178",
+    "@types/node": "16.11.19",
+    "webpack": "5.65.0"
```
Comment (kellymears) [inline sources/@roots/eslint-config-bud/package.json]: pretty much everything except `skypack` is unnecessary

File: sources/@roots/eslint-config-bud/package.json [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    "eslint-plugin-jsx-a11y": "^6.5.1",
+    "eslint-plugin-react": "^7.27.1",
+    "eslint-plugin-react-hooks": "^4.3.0",
+    "tslib": "^2.3.0"
+  },
+  "peerDependencies": {
+    "@babel/eslint-parser": "^7.16.5",
+    "@typescript-eslint/eslint-plugin": "^5.10.1",
+    "@typescript-eslint/parser": "5.10.1",
+    "eslint": "^7.32.0 || ^8.2.0",
```
Comment (kellymears) [inline sources/@roots/eslint-config-bud/package.json]: i followed airbnb's general semver ranges for these

**Comment by swalkinshaw** [comment]
would this package be specific to bud at all?

**Comment by kellymears** [comment]
> would this package be specific to bud at all?

@swalkinshaw it doesn't need to be, is my thinking.they're just generic extensions of eslint:recommended.

in the case of them being non-specific it can be `@roots/eslint-config`. in the case of not wanting that it can be `@roots/eslint-config-bud`. It doesn't have to live in the monorepo but it's kind of convenient.

**Comment by kellymears** [comment]
it's a lot easier for people to contribute to this if it's all in one package, is my thinking. right now it's spread out to the point where it was annoying for me to track it all down.

**Comment by swalkinshaw** [comment]
That's fine 👍  but yeah I wouldnt have `bud` in the name then

**Review by retlehs — APPROVED** [review]

**Pull request merged**